app/models/link_report.rb

```diff
@@ -1,3 +1,32 @@
 class LinkReport < ApplicationRecord
   belongs_to :step
+
+  def create_record
+    batch_response = request_batch_of_links
+    if batch_response.nil?
+      self.destroy
```

Comment:
I don't think you need to do an explicit destroy.
You could check if `batch_response.present?` and only `save` the record if it is.

Reply:
I've done this explicitly because I think explicit is better than implicit, even if it's not technically necessary.